src/AdventureGoldGovernance.sol

```diff
@@ -83,11 +87,24 @@ contract AdventureGoldGovernance is IERC6372, ERC20Permit, ERC20Votes {
         emit Withdrawal(msg.sender, amount);
     }
 
-    /// @notice Only transfers to/from this contract are allowed
+    function transfer(address to, uint256 value) public override(ERC20) returns (bool) {
+        revert NonTransferable();
+    }
+
+    function transferFrom(address from, address to, uint256 value) public override(ERC20) returns (bool) {
+        revert NonTransferable();
+    }
```

Comment:
Thoughts on reverting for the approve function as well?

Reply:
There would never be a need for an approval right since this is a non-transferable token (besides mint and burn).